paimon-core/src/main/java/org/apache/paimon/iceberg/migrate/IcebergMigrateHadoopMetadata.java

```diff
@@ -0,0 +1,111 @@
+import org.apache.paimon.utils.Preconditions;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/** Get iceberg table latest snapshot metadata in hadoop. */
+public class IcebergMigrateHadoopMetadata implements IcebergMigrateMetadata {
+    private static final Logger LOG = LoggerFactory.getLogger(IcebergMigrateHadoopMetadata.class);
+
+    private static final String VERSION_HINT_FILENAME = "version-hint.text";
+    private static final String ICEBERG_WAREHOUSE = "iceberg_warehouse";
+
+    private final FileIO fileIO;
+    private final Identifier icebergIdentifier;
+    private final Options icebergOptions;
+
+    private Path icebergLatestMetaVersionPath;
+    private IcebergPathFactory icebergMetaPathFactory;
+
+    public IcebergMigrateHadoopMetadata(
+            Identifier icebergIdentifier, FileIO fileIO, Options icebergOptions) {
+        this.fileIO = fileIO;
+        this.icebergIdentifier = icebergIdentifier;
+        this.icebergOptions = icebergOptions;
+    }
+
+    @Override
+    public IcebergMetadata icebergMetadata() {
+        Preconditions.checkArgument(
+                icebergOptions.get(ICEBERG_WAREHOUSE) != null,
+                "'iceberg_warehouse' is null. "
+                        + "In hadoop-catalog, you should explicitly set this argument for finding iceberg metadata.");
+        this.icebergMetaPathFactory =
+                new IcebergPathFactory(
+                        new Path(
+                                icebergOptions.get(ICEBERG_WAREHOUSE),
+                                new Path(
+                                        String.format(
+                                                "%s/%s/metadata",
+                                                icebergIdentifier.getDatabaseName(),
+                                                icebergIdentifier.getTableName()))));
+        long icebergLatestMetaVersion = getIcebergLatestMetaVersion();
+
+        this.icebergLatestMetaVersionPath =
+                icebergMetaPathFactory.toMetadataPath(icebergLatestMetaVersion);
+        LOG.info(
+                "iceberg latest snapshot metadata file location: {}", icebergLatestMetaVersionPath);
+
+        return IcebergMetadata.fromPath(fileIO, icebergLatestMetaVersionPath);
+    }
+
+    @Override
+    public String icebergLatestMetadataLocation() {
+        return icebergLatestMetaVersionPath.toString();
+    }
+
+    @Override
+    public void deleteOriginTable() {
+        Path tablePath = icebergMetaPathFactory.metadataDirectory().getParent();
+        LOG.info("Iceberg table path to be deleted:{}", tablePath);
+        try {
+            if (fileIO.isDir(tablePath)) {
+                fileIO.deleteDirectoryQuietly(tablePath);
+            }
+        } catch (IOException e) {
+            LOG.warn(
+                    "exception occurred when deleting origin table, exception message:{}",
+                    e.getMessage());
```

Comment:
```suggestion
            LOG.warn(
                    "exception occurred when deleting origin table", e);
```

In this way the stack trace is also logged.